source/dub/compilers/gdc.d

```diff
@@ -54,31 +54,16 @@ class GDCCompiler : Compiler {
 
 	@property string name() const { return "gdc"; }
 
-	enum gdcVersionRe1 = `GDC\s+(\d+\.\d+\.\d+[A-Za-z0-9.+-]*)`;
-	enum gdcVersionRe2 = `^GNU D \(.*?\) version (\d+\.\d+\.\d+[A-Za-z0-9.+-]*)`;
-	enum gdcVersionRe3 = `^gcc version (\d+\.\d+\.\d+[A-Za-z0-9.+-]*)`;
-
-	unittest {
-		import std.algorithm : equal, map;
-		import std.range : only;
-		import std.regex : matchFirst, regex;
-		auto probe = `
-Target: x86_64-pc-linux-gnu
-Thread model: posix
-gcc version 8.2.1 20180831 (GDC 8.2.1 based on D v2.068.2 built with ISL 0.20 for Arch Linux)
-GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
-GNU D (GDC 8.2.1 based on D v2.068.2 built with ISL 0.20 for Arch Linux) version 8.2.1 20180831 (x86_64-pc-linux-gnu)
-GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
-binary    /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/cc1d
-version   v2.068.2
-predefs   GNU D_Version2 LittleEndian GNU_DWARF2_Exceptions GNU_StackGrowsDown GNU_InlineAsm D_LP64 D_PIC assert all X86_64 D_HardFloat Posix linux CRuntime_Glibc
-`;
-		auto vers = only(gdcVersionRe1, gdcVersionRe2, gdcVersionRe3)
-			.map!(re => matchFirst(probe, regex(re, "m")))
-			.filter!(c => c && c.length > 1)
-			.map!(c => c[1]);
-
-		assert(vers.equal([ "8.2.1", "8.2.1", "8.2.1" ]));
+	string determineVersion(string compiler_binary, string verboseOutput)
+	{
+		const args = [
+			compiler_binary,
+			"-dumpfullversion",
+			"-dumpversion"
```

Comment:
Does this work with older GDC versions too?

Reply:
Surprisingly no, all GDC versions just ignore even invalid flags if `-dumpfullversion` or `-dumpversion` is present in the command line.

`-dumpversion` returns only the major version since gcc-7.

See here an output example: https://github.com/smanders/externpro/issues/160#issuecomment-355154694

Here is gcc-5 and gdc-8 output:

> $ gcc -dumpfullversion -dumpversion
> 5.5.0
> $ gdc -dumpfullversion -dumpversion
> 8.2.0

Reply:
Or to be more precise `-dumpversion` outputs the version used in the directory structure, in the directories like "/opt/gdc/include/c++/**8.2.0**/algorithm" and it can be "/opt/gdc/include/c++/**8**/algorithm" if `--with-gcc-major-version-only` is given at build time.